debian/helpers/build-manpages.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+# This script generates all the manpages based on the upstream scripts. They
+# use to be executed during build time, but now it requires the download of
+# some modules not in the vendor directory (upstream removed dependencies just
+# for docs).
+# So execute this script right after importing a new upstream version to update
+# the manpages in debian/manpages.
+
+set -eux
+
+# Create temporary directory to build manpages
+build_dir=$(mktemp -d -t docker-cli-docsgen.XXXXXXXXXX)
+
+# Define function to remove the build directory when the script exits
+function clean {
+  rm -rf "${build_dir}"
+}
+trap clean EXIT
+
+# Copy the CLI code to the build directory
+cp -r cli/ "${build_dir}"
+pushd "${build_dir}"/cli
+
+# Add Go 1.20 to the $PATH to be used instead of the default
+# (in case the default is not 1.20).
+PATH=/usr/lib/go-1.20/bin:"${PATH}"
```

Comment:
This will need to be adjusted whenever a new Go version comes out, right (or, best case scenario, when Go 1.20 is removed from the archive)?  Isn't there a way to automatically detected the latest Go version installed and populate `PATH` with it?

Reply:
You are right, this will need to be updated manually. I tried to force Go 1.20 here because there is the case where the user will have just the default installed, and this version might not be enough. I am trying to lock to this version since we are building Docker everywhere with Go 1.20. What I think we could do better is first check if `golang-1.20-go` is installed and then update the `$PATH`, if it is not installed raise an error saying that it requires `golang-1.20-go`. WDYT?

Reply:
Hm, but that would still require manual intervention when Go 1.20 is not available/enough to build the manpages.  I think what you can do is check for Go's version, verify if it's at least 1.20, and error out (with a descriptive message) if it isn't.  Then don't worry about adjust `PATH` at all; leave that to the developer who's running the script to do.  This is still not perfect because it will require adjustments if Go 1.20 becomes incapable of generating the manpages, but I'm assuming that this will not happen often.